node/pkg/publicrpc/publicrpcserver.go

```diff
@@ -75,6 +76,17 @@ func (s *PublicrpcServer) GetSignedVAA(ctx context.Context, req *publicrpcv1.Get
 	addr := vaa.Address{}
 	copy(addr[:], address)
 
+	// Chain input validation
+	chainId, err := vaa.ChainIDFromInt(uint16(req.MessageId.EmitterChain))
+	if err != nil || chainId == vaa.ChainIDUnset {
+		return nil, status.Error(codes.InvalidArgument, "invalid chain specified")
+	}
```

Comment:
The UX of `vaa.ChainIDFromInt()` really tickles my code smell spidey sense here. If the return'd `chainId` here is 0, both of these conditions will be true:

```
if err != nil || chainId == vaa.ChainIDUnset {
```

Due to: https://github.com/certusone/wormhole/blob/ec0a32b0a82a0c0d4906bdea45e2ac634fe77db2/node/pkg/vaa/structs.go#L179

So you have a duplicate return AND a duplicate check, both of which are unnecessary. Simplify, simplify, simplify!

If I understand this PR correctly, you're verifying the passed in chainId is valid. If that's correct, this is just a boolean check and doesn't need an error.

So the above code could become simply

```suggestion
	if !vaa.IsValidChainId(uint16(req.MessageId.EmitterChain) {
		return nil, status.Error(codes.InvalidArgument, "invalid chain specified")
	}
```